subsys/logging/log_core.c

```diff
@@ -28,6 +28,161 @@ static bool initialized;
 
 static timestamp_get_t timestamp_func;
 
+/* MOJ KOD */
+
+#ifdef CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE
+	struct LOG_FRONTEND_PIN_NUMBERS log_frontend_pin_nums = {
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+1,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+2,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+3,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+4,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+5,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+6,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+7,
+		CONFIG_LOG_FRONTEND_PIN_CLK
+		};
+#else
+	struct LOG_FRONTEND_PIN_NUMBERS log_frontend_pin_nums = {
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_FIRST_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_SECOND_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_THIRD_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_FOURTH_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_FIFTH_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_SIXTH_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_SEVENTH_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_EIGHTH_BIT,
+		CONFIG_LOG_FRONTEND_PIN_CLK
+		};
+#endif
+
+uint32_t mask_of_pins_log = 0x000005FE;
+
+uint32_t create_mask_gpio(struct LOG_FRONTEND_PIN_NUMBERS * pin_nums, uint8_t message)
+{
+	uint32_t mask = 0;
+
+	if(IS_ENABLED(CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE))
+	{
+	mask = mask | (message << pin_nums->BIT0);
+	}
+	else
+	{
+	mask = (mask | ((((message >> 0)  & 0x01) << (pin_nums->BIT0)))
+		 | ((((message >> 1)  & 0x01) << (pin_nums->BIT1)))
+		 | ((((message >> 2)  & 0x01) << (pin_nums->BIT2)))
+		 | ((((message >> 3)  & 0x01) << (pin_nums->BIT3)))
+		 | ((((message >> 4)  & 0x01) << (pin_nums->BIT4)))
+		 | ((((message >> 5)  & 0x01) << (pin_nums->BIT5)))
+		 | ((((message >> 6)  & 0x01) << (pin_nums->BIT6)))
+		 | ((((message >> 7)  & 0x01) << (pin_nums->BIT7))));
+	}
+	return mask;
+}
+
+uint32_t create_mask_gpio_of_pins(struct LOG_FRONTEND_PIN_NUMBERS * pin_nums)
+{
+	uint32_t mask_of_pins = create_mask_gpio(pin_nums, 255);
+	mask_of_pins = mask_of_pins | (0x00000001 << (pin_nums->CLK));
+	return mask_of_pins;
+}
+
+void send_message_gpio_8b(uint8_t message, struct LOG_FRONTEND_PIN_NUMBERS * pin_nums, uint32_t mask_of_pins)
+{
+	uint32_t mask_to_set = create_mask_gpio(pin_nums, message);
+	nrf_gpio_port_out_set(LOG_FRONTEND_GPIO_REG_PTR, mask_to_set);
```

Comment:
maybe more optimal way would be to do that like this:
create a constant mask (e.g. if pins 20-28 are used mask like 0x1ff00000),
first clear all nine pins, then set all nine pins.